api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -470,14 +470,19 @@ public boolean hasUserRunOutOfInitialCredits(DbUser user) {
     return usage >= limit;
   }
 
+  private boolean shouldCheckApproachingExpirationNotification(DbUserInitialCreditsExpiration exp) {
+    return !checkEnableUnlinkBillingForInitialCreditsFlag()
+        && exp.getApproachingExpirationNotificationTime() == null;
```

Comment:
This won't send any emails once the feature flag is flipped because `!checkEnableUnlinkBillingForInitialCreditsFlag()` will evaluate to false. This should give us the correct behavior:

```
return checkEnableUnlinkBillingForInitialCreditsFlag()
        || exp.getApproachingExpirationNotificationTime() == null;
```